crates/re_sdk/src/log_sink.rs

```diff
@@ -6,47 +9,31 @@ pub trait LogSink: Send + Sync + 'static {
     fn send(&self, msg: LogMsg);
 
     /// Send all these log messages.
+    #[inline]
     fn send_all(&self, messages: Vec<LogMsg>) {
         for msg in messages {
             self.send(msg);
         }
     }
 
     /// Drain all buffered [`LogMsg`]es and return them.
+    ///
+    /// Only applies to sinks that maintain a backlog.
+    #[inline]
```

Comment:
The fact that all sinks implement this but don't support it strikes me as awkward.  It is possible to be explicit about tracking when our Sink is a BufferedSink and only implement drain_backlog there?

At a minimum maybe signature should be `Option<Vec<LogMsg>>` to disambiguate between "The sink doesn't support a backlog" and "The backlog was empty."

Reply:
I mean sure we could return an option but I don't see how that'd be an improvement? We can always add later on it if we ever need to disambiguate between `no backlog` vs. `empty backlog`, but there's no such need at the moment.

Reply:
> if we ever need to disambiguate between no backlog vs. empty backlog

Yeah, we can talk about this more concretely in a future context. I'm thinking about cases like: "I tried to use a Tcp sink, but the tcp end-point wasn't there, so now instead I try to switch my sink to a FileSink, expecting it to drain the backlog of buffered tcp messages, but even though the tcp sink told me there was no backlog, I actually just lost my data"